Glassfish 7.0.15 Linux java.lang.ClassNotFoundException: org.glassfish.jersey.moxy.json.internal.MoxyJsonAutoDiscoverable

## Environment Details
* GlassFish Version (and build number): 7.0.15
* JDK version: jdk-21.0.3
* OS: Linux (Kernel 6.1.0-18-amd64)

----------

## Problem Description
```
[2024-06-27T08:40:33.016409Z] [GF 7.0.15] [SEVERE] [] [jakarta.enterprise.system.core] [tid: _ThreadID=67 _ThreadName=admin-listener(3)] [levelValue: 1000] [[
  org.glassfish.jersey.moxy.json.internal.MoxyJsonAutoDiscoverable
java.lang.ClassNotFoundException: org.glassfish.jersey.moxy.json.internal.MoxyJsonAutoDiscoverable
	at org.apache.felix.framework.Felix.loadBundleClass(Felix.java:2102)
	at org.apache.felix.framework.BundleImpl.loadClass(BundleImpl.java:986)
	at org.jvnet.hk2.osgiadapter.OSGiModuleImpl$4$1.run(OSGiModuleImpl.java:467)
	at org.jvnet.hk2.osgiadapter.OSGiModuleImpl$4$1.run(OSGiModuleImpl.java:464)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:571)
	at org.jvnet.hk2.osgiadapter.OSGiModuleImpl$4.loadClass(OSGiModuleImpl.java:464)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	at com.sun.enterprise.v3.server.APIClassLoaderServiceImpl$APIClassLoader.loadClass(APIClassLoaderServiceImpl.java:244)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:580)
	at org.glassfish.common.util.GlassfishUrlClassLoader.loadClass(GlassfishUrlClassLoader.java:115)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:534)
	at java.base/java.lang.Class.forName(Class.java:513)
	at org.glassfish.jersey.internal.util.ReflectionHelper$7.run(ReflectionHelper.java:382)
	at org.glassfish.jersey.internal.util.ReflectionHelper$7.run(ReflectionHelper.java:377)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:571)
	at org.glassfish.jersey.internal.ServiceFinder$AbstractLazyIterator.hasNext(ServiceFinder.java:556)
	at org.glassfish.jersey.internal.ServiceFinder.toClassArray(ServiceFinder.java:397)
	at org.glassfish.jersey.internal.AbstractServiceFinderConfigurator.loadImplementations(AbstractServiceFinderConfigurator.java:63)
	at org.glassfish.jersey.internal.AutoDiscoverableConfigurator.init(AutoDiscoverableConfigurator.java:49)
	at org.glassfish.jersey.server.ApplicationHandler.lambda$initialize$0(ApplicationHandler.java:306)
	at java.base/java.util.Arrays$ArrayList.forEach(Arrays.java:4305)
	at org.glassfish.jersey.server.ApplicationHandler.initialize(ApplicationHandler.java:306)
	at org.glassfish.jersey.server.ApplicationHandler.<init>(ApplicationHandler.java:273)
	at org.glassfish.jersey.server.ApplicationHandler.<init>(ApplicationHandler.java:260)
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.<init>(GrizzlyHttpContainer.java:310)
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainerProvider.createContainer(GrizzlyHttpContainerProvider.java:37)
	at org.glassfish.jersey.server.ContainerFactory.createContainer(ContainerFactory.java:58)
	at org.glassfish.admin.rest.adapter.RestAdapter.getJerseyContainer(RestAdapter.java:282)
	at org.glassfish.admin.rest.adapter.RestAdapter.exposeContext(RestAdapter.java:275)
	at org.glassfish.admin.rest.adapter.RestAdapter.service(RestAdapter.java:151)
	at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:425)
	at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:144)
	at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:174)
	at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:153)
	at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:196)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:88)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:246)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:178)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:118)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:96)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:51)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:510)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:82)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:83)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:101)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:535)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:515)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: org.osgi.framework.BundleException: Unable to resolve org.glassfish.jersey.media.jersey-media-moxy [189](R 189.0): missing requirement [org.glassfish.jersey.media.jersey-media-moxy [189](R 189.0)] osgi.wiring.package; (&(osgi.wiring.package=org.eclipse.persistence.jaxb)(version>=4.0.0)(!(version>=5.0.0))) Unresolved requirements: [[org.glassfish.jersey.media.jersey-media-moxy [189](R 189.0)] osgi.wiring.package; (&(osgi.wiring.package=org.eclipse.persistence.jaxb)(version>=4.0.0)(!(version>=5.0.0)))]
	at org.apache.felix.framework.Felix.resolveBundleRevision(Felix.java:4398)
	at org.apache.felix.framework.Felix.loadBundleClass(Felix.java:2095)
	... 49 more
]]

[2024-06-27T08:40:33.018261Z] [GF 7.0.15] [SEVERE] [] [jakarta.enterprise.system.core] [tid: _ThreadID=67 _ThreadName=admin-listener(3)] [levelValue: 1000] [[
  
Failed loading class org.glassfish.jersey.moxy.json.internal.MoxyJsonAutoDiscoverable by API Class Loader

Unable to resolve
    org.glassfish.jersey.media.jersey-media-moxy [189]
    missing requirement
        &(package = org.eclipse.persistence.jaxb) (version >= 4.0.0) (!(version >= 5.0.0))

]]

[2024-06-27T08:40:33.201064Z] [GF 7.0.15] [SEVERE] [] [jakarta.enterprise.system.core] [tid: _ThreadID=67 _ThreadName=admin-listener(3)] [levelValue: 1000] [[
  org.glassfish.jersey.jsonp.internal.JsonProcessingAutoDiscoverable
java.lang.ClassNotFoundException: org.glassfish.jersey.jsonp.internal.JsonProcessingAutoDiscoverable
	at org.apache.felix.framework.Felix.loadBundleClass(Felix.java:2102)
	at org.apache.felix.framework.BundleImpl.loadClass(BundleImpl.java:986)
	at org.jvnet.hk2.osgiadapter.OSGiModuleImpl$4$1.run(OSGiModuleImpl.java:467)
	at org.jvnet.hk2.osgiadapter.OSGiModuleImpl$4$1.run(OSGiModuleImpl.java:464)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:571)
	at org.jvnet.hk2.osgiadapter.OSGiModuleImpl$4.loadClass(OSGiModuleImpl.java:464)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	at com.sun.enterprise.v3.server.APIClassLoaderServiceImpl$APIClassLoader.loadClass(APIClassLoaderServiceImpl.java:244)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:580)
	at org.glassfish.common.util.GlassfishUrlClassLoader.loadClass(GlassfishUrlClassLoader.java:115)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:534)
	at java.base/java.lang.Class.forName(Class.java:513)
	at org.glassfish.jersey.internal.util.ReflectionHelper$7.run(ReflectionHelper.java:382)
	at org.glassfish.jersey.internal.util.ReflectionHelper$7.run(ReflectionHelper.java:377)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:571)
	at org.glassfish.jersey.internal.ServiceFinder$AbstractLazyIterator.hasNext(ServiceFinder.java:556)
	at org.glassfish.jersey.internal.ServiceFinder.toClassArray(ServiceFinder.java:397)
	at org.glassfish.jersey.model.internal.CommonConfig.configureAutoDiscoverableProviders(CommonConfig.java:600)
	at org.glassfish.jersey.server.ResourceConfig.configureAutoDiscoverableProviders(ResourceConfig.java:819)
	at org.glassfish.jersey.server.ApplicationHandler.initialize(ApplicationHandler.java:338)
	at org.glassfish.jersey.server.ApplicationHandler.lambda$initialize$1(ApplicationHandler.java:309)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
	at org.glassfish.jersey.internal.Errors.processWithException(Errors.java:232)
	at org.glassfish.jersey.server.ApplicationHandler.initialize(ApplicationHandler.java:308)
	at org.glassfish.jersey.server.ApplicationHandler.<init>(ApplicationHandler.java:273)
	at org.glassfish.jersey.server.ApplicationHandler.<init>(ApplicationHandler.java:260)
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.<init>(GrizzlyHttpContainer.java:310)
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainerProvider.createContainer(GrizzlyHttpContainerProvider.java:37)
	at org.glassfish.jersey.server.ContainerFactory.createContainer(ContainerFactory.java:58)
	at org.glassfish.admin.rest.adapter.RestAdapter.getJerseyContainer(RestAdapter.java:282)
	at org.glassfish.admin.rest.adapter.RestAdapter.exposeContext(RestAdapter.java:275)
	at org.glassfish.admin.rest.adapter.RestAdapter.service(RestAdapter.java:151)
	at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:425)
	at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:144)
	at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:174)
	at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:153)
	at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:196)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:88)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:246)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:178)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:118)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:96)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:51)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:510)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:82)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:83)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:101)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:535)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:515)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: org.osgi.framework.BundleException: Unable to resolve org.glassfish.jersey.media.jersey-media-json-processing [188](R 188.0): missing requirement [org.glassfish.jersey.media.jersey-media-json-processing [188](R 188.0)] osgi.wiring.package; (&(osgi.wiring.package=org.eclipse.parsson.media)(version>=1.1.0)(!(version>=2.0.0))) [caused by: Unable to resolve org.eclipse.parsson.media [247](R 247.0): missing requirement [org.eclipse.parsson.media [247](R 247.0)] osgi.wiring.package; (&(osgi.wiring.package=jakarta.annotation)(version>=3.0.0)(!(version>=4.0.0)))] Unresolved requirements: [[org.glassfish.jersey.media.jersey-media-json-processing [188](R 188.0)] osgi.wiring.package; (&(osgi.wiring.package=org.eclipse.parsson.media)(version>=1.1.0)(!(version>=2.0.0)))]
	at org.apache.felix.framework.Felix.resolveBundleRevision(Felix.java:4398)
	at org.apache.felix.framework.Felix.loadBundleClass(Felix.java:2095)
	... 52 more
]]

[2024-06-27T08:40:33.202443Z] [GF 7.0.15] [SEVERE] [] [jakarta.enterprise.system.core] [tid: _ThreadID=67 _ThreadName=admin-listener(3)] [levelValue: 1000] [[
  
Failed loading class org.glassfish.jersey.jsonp.internal.JsonProcessingAutoDiscoverable by API Class Loader

Unable to resolve
    org.glassfish.jersey.media.jersey-media-json-processing [188]
    missing requirement
        &(package = org.eclipse.parsson.media) (version >= 1.1.0) (!(version >= 2.0.0))
        caused by:
            Unable to resolve
                org.eclipse.parsson.media [247]
                missing requirement
                    &(package = jakarta.annotation) (version >= 3.0.0) (!(version >= 4.0.0)))]

]]
```

## Steps to reproduce
Download glassfish-7.0.15.zip
Start domain1 and open http://localhost:4848/

This bug is only under linux, glassfish 7.0.15 start normaly under Windows with same JDK Version
Glassfish 7.0.14 has not this bug


```
cd glassfih
mvn clean install -Pstaging,fastest -T4C
mvn clean install -Pstaging -pl :application-tests
less appserver/tests/application/target/glassfish7/glassfish/domains/domain1/logs/server.log
```

--- a/appserver/featuresets/web/pom.xml
+++ b/appserver/featuresets/web/pom.xml
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!--
 
-    Copyright (c) 2021, 2023 Contributors to Eclipse Foundation.
+    Copyright (c) 2021, 2024 Contributors to Eclipse Foundation.
     Copyright (c) 2015, 2022 Oracle and/or its affiliates. All rights reserved.
 
     This program and the accompanying materials are made available under the
@@ -164,6 +164,9 @@
           </dependency>
 
         <!-- Jersey -->
+        <!-- Since MOXy has been removed from GlassFish,
+             the Jersey-MOXy integration module should not be there.
+             See https://github.com/eclipse-ee4j/glassfish/issues/24749
         <dependency>
             <groupId>org.glassfish.jersey.media</groupId>
             <artifactId>jersey-media-moxy</artifactId>
@@ -174,6 +177,7 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        -->
         <dependency>
             <groupId>org.glassfish.jersey.ext.cdi</groupId>
             <artifactId>jersey-cdi1x</artifactId>

